### PR TITLE
Fix local start for UD

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Fix local start for UD

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/20897

### Is it tested? How?
1.  checkout https://github.com/eclipse-che/che-dashboard/tree/CHE-20897
2. `yarn`
3. `yarn build`
4. `yarn start:prepare`
5. `yarn start`

### CODE REVIEW

File: packages/dashboard-backend/src/local-run/dexHelper.ts [hunk 1]
```diff
@@ -96,10 +103,30 @@ export function addDexProxy(server: FastifyInstance) {
     rewritePrefix: '/auth',
     disableCache: true,
   });
+}
+
+export function addAutoresizeHooks(server: FastifyInstance) {
```
Comment (akurinnoy) [inline packages/dashboard-backend/src/local-run/dexHelper.ts:108]: I seem it's a typo

```suggestion
export function addAuthorizationHooks(server: FastifyInstance) {
```

File: packages/dashboard-backend/src/local-run/dexHelper.ts [hunk 2]
```diff
@@ -96,10 +103,30 @@ export function addDexProxy(server: FastifyInstance) {
...
+  server.addHook('onResponse', (request, reply, done) => {
+    if (
+      (request.url.startsWith('/api/') || request.url.startsWith('/dashboard/api/')) &&
+      request.method === 'GET' &&
+      reply.statusCode == 401
+    ) {
+      if (reply.statusCode === 401) {
+        process.env.CLUSTER_ACCESS_TOKEN = '';
+      }
```
Comment (akurinnoy) [inline packages/dashboard-backend/src/local-run/dexHelper.ts:117]: This condition is always true:

```suggestion
      process.env.CLUSTER_ACCESS_TOKEN = '';
```

File: packages/dashboard-backend/src/local-run/dexHelper.ts [hunk 3]
```diff
@@ -96,10 +103,30 @@ export function addDexProxy(server: FastifyInstance) {
...
     disableCache: true,
   });
+}
+
+export function addAutoresizeHooks(server: FastifyInstance) {
+  server.addHook('onResponse', (request, reply, done) => {
+    if (
+      (request.url.startsWith('/api/') || request.url.startsWith('/dashboard/api/')) &&
+      request.method === 'GET' &&
+      reply.statusCode == 401
```
Comment (akurinnoy) [inline packages/dashboard-backend/src/local-run/dexHelper.ts:113]:
```suggestion
      reply.statusCode === 401
```

**Review by akurinnoy — APPROVED** [review]
Tested, local development flow works as expected!

File: packages/dashboard-backend/src/local-run/dexHelper.ts (hunk 2 above)
Comment (olexii4) [reply packages/dashboard-backend/src/local-run/dexHelper.ts]: Good catch. I have fixed it.

File: packages/dashboard-backend/src/local-run/dexHelper.ts (hunk 3 above)
Comment (olexii4) [reply packages/dashboard-backend/src/local-run/dexHelper.ts]: Good catch. I have fixed it.

File: packages/dashboard-backend/src/local-run/dexHelper.ts (hunk 1 above)
Comment (olexii4) [reply packages/dashboard-backend/src/local-run/dexHelper.ts]: I have fixed it.

**Pull request merged**